### PR TITLE
Strip fields in /labels endpoints

### PR DESCRIPTION
**Pull request by naorgu**
Currently, `/labels` endpoint does not strip the labels passed in the `Restrict-By-Tags-JSON` header.

This PR fixes it by stripping the labels as part of the json rendering (`RenderListTagResultsJSON`). The labels to strip were added to the `RenderSeriesMetadataOptions` struct, since this struct is being used by other response writers which should support strip anyhow - this can be done in future PR.

Also contains a test.

### CODE REVIEW

**Comment by asafm** [comment]
@wesleyk Is it possible for someone to review this PR?

**Review by arnikola — COMMENTED** [review]
Hm, we should be filtering out these tags at the fanout storage level here; https://github.com/chronosphereio/m3/blob/master/src/query/storage/fanout/storage.go#L514, are we not seeing that happen in practice?

**Comment by asafm** [comment]
@arnikola ping 2 :)

**Comment by asafm** [comment]
@gibbscullen Can you help with getting feedback for above comment?

**Comment by gibbscullen** [comment]
> @gibbscullen Can you help with getting feedback for above comment?

We will have someone look into this

**Comment by arnikola** [comment]
Oh super sorry, have definitely not been up to date on my notifications :(
Good catch that we're not applying the filter if we're running on a single store; would something like this in `fanoutStorage.CompleteTags` work for you? Think we can get away without having to re-invent the wheel :)

```golang
        if len(stores) == 1 {
		result, err := stores[0].CompleteTags(ctx, query, options)
		if err != nil {
			return nil, err
		}
		res := applyOptions(*result, options)
		return &res, nil
	}
```

**Comment by asafm** [comment]
@arnikola Thanks - looks good. I'll add it to the PR and ping you.
I tried pinging through the community Slack but couldn't find someone matching your github user there.

**Comment by arnikola** [comment]
Ah just wanted to check if we wanted to do this in the shortcircuit of `fanoutStorage.CompleteTags` as discussed above? I'm the plain "Artem" on the slack, just set my subtitle to `arnikola` if you want to chat about this 👍

**Comment by asafm** [comment]
@arnikola Thanks for the feedback. I deleted the previous fix and added a new fix per last comments. Also added test that fails prior to my fix.

File: src/x/test/string.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package test
+
+func BytesArray(strings ...string) [][]byte {
```
Comment (arnikola) [inline src/x/test/string.go:23]: nit: probably fine to just keep this helper function in the test itself, doesn't seem to be used elsewhere. If you want to keep it here though, mind updating the copyright header to read `2021`?

Comment (asafm) [reply src/x/test/string.go:23]: I will update the header.
I just saw so many copies of the same functions doing the same thing, I thought it made sense to have it in a single place.

**Comment by asafm** [comment]
@arnikola I've change copyrights year and changed location of single line comment to more appropriate place

**Comment by asafm** [comment]
@arnikola I fixed the lint - you see the code changes to make sure they are ok. The tests are failing regardless of my change - I took the failures of test_harness as an example - re-ran on my laptop and they passed.

**Comment by asafm** [comment]
@arnikola Branch is green

**Comment by asafm** [comment]
@arnikola The build jobs are highly flaky - you can see previous one was green

**Comment by asafm** [comment]
I don't have permissions to re-run them individually

**Comment by asafm** [comment]
@arnikola Finally all green!

**Pull request merged**